src/store/ws/main-table.ts

```diff
@@ -0,0 +1,228 @@
+import { makeAutoObservable, reaction, runInAction, toJS } from 'mobx'
+
+import datasetStore from '@store/dataset'
+import filterStore from '@store/filter'
+import filterPresetsStore from '@store/filter-presets'
+import variantStore from '@store/ws/variant'
+import zoneStore from '@store/ws/zone'
+import {
+  TCondition,
+  TItemsCount,
+} from '@service-providers/common/common.interface'
+import datasetProvider from '@service-providers/dataset-level/dataset.provider'
+import { ITabReport } from '@service-providers/dataset-level/dataset-level.interface'
+import wsDatasetProvider from '@service-providers/ws-dataset-support/ws-dataset-support.provider'
+import { IWsListQuery, WsListAsyncStore } from './ws-list.async.store'
+
+const INCREASE_INDEX = 50
+
+export class MainTable {
+  private readonly wsList = new WsListAsyncStore()
+  tabReport: ITabReport[] = []
+
+  selectedVariantNumber?: number
+  variantsAmount = 0
+  indexTabReport = 0
+  indexFilteredNo = 0
+
+  isLoadingTabReport = false
+  isFetchingMore = false
+  isFilterDisabled = false
+  reportsLoaded = false
+
+  memorizedConditions:
+    | {
+        conditions: ReadonlyArray<TCondition>
+        activePreset: string
+        zone: any[]
+      }
+    | undefined = undefined
+
+  constructor() {
+    makeAutoObservable(this)
+
+    reaction(
+      () => this.wsListQuery,
+      query => {
+        if (datasetStore.isXL) return
+
+        if (!datasetStore.datasetName) {
+          this.wsList.reset()
+        } else {
+          this.wsList.setQuery(query)
+        }
+      },
+    )
+
+    reaction(
+      () => this.wsRecords,
+      () => {
+        this.indexFilteredNo = 0
+        this.fetchFilteredTabReportAsync()
+      },
+    )
+  }
+
+  private get wsListQuery(): IWsListQuery {
+    const conditions = !filterPresetsStore.activePreset
+      ? filterStore.conditions
+      : []
+    return {
+      datasetName: datasetStore.datasetName,
+      filter: filterPresetsStore.activePreset,
+      conditions,
+      zone: zoneStore.zone,
+    }
+  }
+
+  public get wsRecords() {
+    return this.wsList.data?.records
+  }
+
+  public get filteredNo(): number[] {
+    return this.wsRecords
+      ? this.wsRecords.map((variant: { no: number }) => variant.no)
+      : []
+  }
+
+  public get statAmount(): TItemsCount | undefined {
+    return filterStore.conditions.length || zoneStore.zone.length
+      ? this.wsList.data?.['filtered-counts']
+      : this.wsList.data?.['total-counts']
```

Comment:
I would suggest to create an adapter for wsDatasetSupportProvider.getWsList to convert kebab-cased props to camel

Reply:
added